cfme/tests/openstack/cloud/test_instances.py

```diff
@@ -176,12 +176,12 @@ def test_list_vms_infra_node(appliance, provider, soft_assert):
     host_collection = appliance.collections.hosts
     # Match hypervisors by IP with count of running VMs
     hvisors = {hv.host_ip: hv.running_vms for hv in provider.mgmt.api.hypervisors.list()}
+    hv_ips = hvisors.keys()
```

Comment:
**required** if the data from the ux doesn't match the backend data and you are certain order is exactly the same,
i think its best to have `hvisors = provider.mgmt.api.hypervisors.list()` in order to ensure the consistent order at the initial declaration

Reply:
also **required** documenting the order expectations, without knowledge of the order of things this test and change looks highly fragile

Reply:
I got your point. looks like my changes worked by accident
The order is not necessarily the same, so I need to find a different  way to tackle this issue

Reply:
I am correcting myself:
The order is preserved, I have checked that several time.
The interesting point is that with **enumerate** (current commit) the order is preserved and test is passing but with **zip**  is not preserved and test fails. final result of test in not an indicator but with knowledge of  order of things it can be a good one